tests/NSubstitute.Analyzers.Tests.Shared/Text/TextParser.cs

```diff
@@ -0,0 +1,182 @@
+﻿// https://github.com/JosefPihrt/Roslynator/blob/master/src/Tests/TestFramework.Core/Text/TextParser.cs
```

Comment:
Should include the commit this is from (https://github.com/JosefPihrt/Roslynator/blob/e18e430246be649c418cc17d393b48d8e1cf7feb/src/Tests/TestFramework.Core/Text/TextParser.cs ?).

I think we also need to keep the copyright notice in tact:

> // Copyright (c) Josef Pihrt. All rights reserved. Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.

May need to edit to be "See License.txt at https://github.com/JosefPihrt/Roslynator for license information". I suggest this:

```csharp
// Src: https://github.com/JosefPihrt/Roslynator/blob/e18e430246be649c418cc17d393b48d8e1cf7feb/src/Tests/TestFramework.Core/Text/TextParser.cs
// Copyright (c) Josef Pihrt. All rights reserved. Licensed under the Apache License, Version 2.0. See License.txt at at https://github.com/JosefPihrt/Roslynator for license information.
// Some modifications made (such as namespace) for NSubstitute.Analyzers project.
```

I'm not sure on the legalities of how this stuff works, but based on section 4. of  http://www.apache.org/licenses/LICENSE-2.0 it looks like the main things we need to do is retain the copyright info, include a prominent notice that we have modified the files, and we must include the original licence.

Reply:
Changed the comment in TextParser.cs as you suggested. When it comes to 
>and we must include the original licence.

Should I include it in our repo? Or the link in the comment is enough?

Reply:
I think the comment should be sufficient?